axelor-base/src/main/java/com/axelor/apps/base/service/pricing/PricingServiceImpl.java

```diff
@@ -150,44 +146,27 @@ protected void pricingFetchFilter(
       bindings.put("company", company);
     }
 
-    if (modelName != null) {
+    if (model != null) {
       filter.append("AND self.concernedModel.name = :modelName ");
-      bindings.put("modelName", modelName);
+      bindings.put("modelName", EntityHelper.getEntityClass(model).getSimpleName());
     }
   }
 
-  protected void appendProductFilter(
-      Product product,
-      ProductCategory productCategory,
-      StringBuilder filter,
-      Map<String, Object> bindings) {
-    StringBuilder productFilter = new StringBuilder();
-    productFilter.append("(");
-    if (product != null) {
-      productFilter.append("self.product = :product ");
-      bindings.put("product", product);
-
-      if (product.getParentProduct() != null) {
-        productFilter.append("OR self.product = :parentProduct ");
-        bindings.put("parentProduct", product.getParentProduct());
-      }
-      if (productCategory != null) {
-        productFilter.append("OR self.productCategory = :productCategory ");
-        bindings.put("productCategory", productCategory);
-      }
-    } else {
-      if (productCategory != null) {
-        productFilter.append("self.productCategory = :productCategory ");
-        bindings.put("productCategory", productCategory);
+  protected List<Pricing> appendFormulaFilter(List<Pricing> pricings, Model model) {
+    Context scriptContext = new Context(Mapper.toMap(model), EntityHelper.getEntityClass(model));
+    ScriptHelper scriptHelper = new GroovyScriptHelper(scriptContext);
+    List<Pricing> filteredPricings = new ArrayList<>();
+    for (Pricing pricing : pricings) {
+      try {
+        if ((Boolean) scriptHelper.eval(pricing.getFormula())) {
```

Comment:
if the user writes a script that evaluates to null, I think this will hrow a NPE, we should avoid this by writing `Boolean.TRUE.equals(...)`